src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -31,6 +31,8 @@ const storageAdapterAllCollections = mongoAdapter => {
     .connect()
     .then(() => mongoAdapter.database.collections())
     .then(collections => {
+      if (!collections || !collections.length) return [];
```

Comment:
Why is this needed? Just to make sure we don't introduce a breaking change here.

Reply:
It's needed because in some of tests due to invalid id, .collections() don't return an array :). So collections.filter crash because it isn't an array :)

Reply:
> collections() don't return an array

When is that the case, for which test?

Reply:
The test are in apple auth and facebook auth. Like 'should use algorithm from key header to verify id_token'.

The problem is thats test spyOn utils.promisify, but now database.collections seems to use it so it's mock this calls also. I search a better solution to mock differently but I don't found anything. With jasmine we can't mock only for onetime, we can't also export the function of facebook adapter and mock it because it's not the same instance. I have no idea. Do you have a proposition @mtrezza  ?

Reply:
´spyOn(util, 'promisify')` looks like bad practice; this a general helper function can be used anywhere and break the test, as it does here. An alternative would be to look at where it's used and mock on a higher level. Which `promisify` is this supposed to mock?

Reply:
We need to mock this function : 
```javascript
const getFacebookKeyByKeyId = async (keyId, cacheMaxEntries, cacheMaxAge) => {
  const client = jwksClient({
    jwksUri: `${TOKEN_ISSUER}/.well-known/oauth/openid/jwks/`,
    cache: true,
    cacheMaxEntries,
    cacheMaxAge,
  });

  const asyncGetSigningKeyFunction = util.promisify(client.getSigningKey);

  let key;
  try {
    key = await asyncGetSigningKeyFunction(keyId);
  } catch (error) {
    throw new Parse.Error(
      Parse.Error.OBJECT_NOT_FOUND,
      `Unable to find matching key for Key ID: ${keyId}`
    );
  }
  return key;
};
```

Reply:
I think we have 3 options:
- mock `getFacebookKeyByKeyId` -> not preferred, as we'd be not testing the code in this function
- mock `client.getSigningKey` -> seems most reasonable
- mock `asyncGetSigningKeyFunction ` -> requires refactoring the code to provide a way to mock it

Reply:
How can I mock `client.getSigningKey` with jasmine (I am not familiar with it). Moreover the client is created in the function so how can we mock the same instance ?

Reply:
I looked but in this case the problem is that the instant of client in the function is different that every function that we mock. So the mocked function is never called. I don't know if in Jasmine it's possible to mock an internal variable of a function

Reply:
Could you tell me what are the tests that are failing, then I'll take a look

Reply:
Almost a big part of the facebook test and apple test in AuthenticationAdapter.spec.js

Reply:
I mean which line in the test specifically?

Reply:
@coratgerl I hope you didn't get discouraged due to the scope of this PR; it would be great if we could get this ready for merge.

Reply:
No of course @mtrezza just so busy last days ahah.

This test for example :) 'should use algorithm from key header to verify id_token'

Reply:
Any solutions for this problem ? It is the only things missing for merge I think

Reply:
I just didn't find time yet to look into it

Reply:
@abhisheksharma010 do you want to look into this? Since you asked to work on this issue

Reply:
@mtrezza I was able to free up a little time on it, I found a solution that seems to fix the problem :)